File: BackEnd/src/models/Event.js
const mongoose = require('mongoose');

const Schema = mongoose.Schema;
const EventSchema = new Schema({
    title: {
        type : String, 
        required: true,
    },
    category: {
        type : String, 
        required: true,
        enum: ['Μουσική', 'Κινηματογράφος', 'Αθλητικά', 'Θέατρο'],
    },
    description: {
        type: String,
        required: true,
    },
    date: {
        type: Date,
        required: true,
    },
    time: {
        type: String,
        required: true,
    },
    remainingTickets: {
        type: Number,
        required: true,
        default: 10,
    },
    ticketPrice: {
        type : Number,
        required: true,
    },
    image: {
        type: String
    },
    city: {
        type: Schema.Types.ObjectId,
        ref: 'City',
        required: true 
    }

});

module.exports = mongoose.model('Event', EventSchema);

File: FrontEnd/app/src/main/java/com/example/app/models/Event.kt
package com.example.app.models

import com.google.gson.annotations.SerializedName

data class Event(
    @SerializedName("_id") val id: String,
    val title: String,
    val category: String,
    val description: String,
    val date: String, // Dates will be represented as Strings here; conversion might be necessary
    val time: String,
    val remainingTickets: Int,
    val ticketPrice: Double,
    val image: String?,
    @SerializedName("city") val cityId: String
)

